platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/SyncPreConsensusEventWriter.java

```diff
@@ -0,0 +1,392 @@
+        }
+
+        if (lastWrittenEvent < eventSequenceNumber) {
+            // We haven't yet written this event, event will be flushed as soon as it is written.
+            flushableEvents.add(eventSequenceNumber);
+            return;
+        }
+
+        // We have written the event to the stream, flush immediately
+        try {
+            currentMutableFile.flush();
+        } catch (final IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        markEventsAsFlushed();
+    }
+
+    /**
+     * Close the output file.
+     */
+    private void closeFile() {
+        try {
+            averageGenerationalSpanUtilization.add(currentMutableFile.getUtilizedGenerationalSpan());
+            currentMutableFile.close();
+
+            // Future work: "compactify" file name here
+
+            fileManager.finishedWritingFile(currentMutableFile);
+            markEventsAsFlushed();
+            currentMutableFile = null;
+
+            // Not strictly required here, but not a bad place to ensure we delete
+            // files incrementally (as opposed to deleting a bunch of files all at once).
+            pruneOldFiles();
+        } catch (final IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    /**
+     * Calculate the generation span for a new file that is about to be created.
+     */
+    private long computeNewFileSpan(final long minimumFileGeneration, final long nextGenerationToWrite) {
+        if (averageGenerationalSpanUtilization.isEmpty()) {
+            return bootstrapGenerationalSpan;
+        }
+
+        final long desiredSpan =
+                (long) (averageGenerationalSpanUtilization.getAverage() * generationalSpanOverlapFactor);
+
+        final long minimumSpan = (nextGenerationToWrite + minimumGenerationalCapacity) - minimumFileGeneration;
+
+        return Math.max(desiredSpan, minimumSpan);
+    }
+
+    /**
+     * Prepare the output stream for a particular event. May create a new file/stream if needed.
+     *
+     * @param eventToWrite
+     * 		the event that is about to be written
+     */
+    private void prepareOutputStream(final EventImpl eventToWrite) throws IOException {
+        if (currentMutableFile != null
+                && (!currentMutableFile.canContain(eventToWrite)
+                        || currentMutableFile.fileSize() * Units.BYTES_TO_MEBIBYTES >= preferredFileSizeMegabytes)) {
+            closeFile();
+        }
+
+        if (currentMutableFile == null) {
+            final long maximumGeneration = minimumGenerationNonAncient
+                    + computeNewFileSpan(minimumGenerationNonAncient, eventToWrite.getGeneration());
+
+            currentMutableFile = fileManager
+                    .getNextFileDescriptor(minimumGenerationNonAncient, maximumGeneration)
+                    .getMutableFile();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() {
```

Comment:
Why do we implement startable if it is not used?

Reply:
This inherits from an interface that is Startable. Another implementation (the asyncronous wrapper) does need to be started.